runtime/vm/resolvesupport.cpp

```diff
@@ -973,11 +973,10 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 		 * since ValueType superclasses can not have fields.
 		 */
 #if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
-		if (J9_IS_J9CLASS_VALUETYPE(resolvedClass)) {
```

Comment:
Why is the value type check and q type check removed here?

Reply:
> I believe we can move the check inside findIndexInFlattenedClassCache().

Added as an assertion.